src/cli/rustup_mode.rs

```diff
@@ -135,6 +141,12 @@ pub fn cli() -> App<'static, 'static> {
                 .long("toolchain")
                 .takes_value(true),
         )
+        .arg(
+            Arg::with_name("release-channel")
+                .help("provide the release channel (e.g. +stable) for toolchain")
+                .index(1)
+                .possible_values(&["+stable", "+nightly", "+beta"]),
```

Comment:
The problem with this is that it doesn't support specific toolchain names, or custom linked toolchains.  This approach simply won't scale.

It may be better to transform the `+foo` into `--toolchain foo` if we can do that and still have clap parse the mutated CLI.